Raise CollectError if import test module fails

omarkohl opened the pull request:
A few tests are still failing and need to be fixed. I will do that and adapt the `CHANGELOG` if the change makes sense.

I honestly don't see much point in fixing #1426 (but did it because it looked simple enough).

Here's a quick checklist that should be present in PRs:

- [X] Target: for bug or doc fixes, target `master`; for new features, target `features`
- [X] Make sure to include one or more tests for your change
- [X] Add yourself to `AUTHORS`
- [x] Add a new entry to the `CHANGELOG` (choose any open position to avoid merge conflicts with other PRs) 

One of the reasons for failing to import the test module is invalid Python
identifiers in the full package path of the test module.

fix #1426

nicoddemus commented:
Thanks! While I agree that this seems like the original user report was a stretch, I don't think it hurts either. :grin:

omarkohl commented:
I improved the error message a little, fixed failing tests and added a CHANGELOG entry.

I force pushed the branch instead of adding commits to avoid cluttering the repository.

nicoddemus commented:
Thanks! The tests are failing because of a minor output detail between Python versions, could you please fix it?

omarkohl commented:
> Thanks! The tests are failing because of a minor output detail between Python versions, could you please fix it?

Yep I'm on it.

nicoddemus commented:
Oh sorry @omarkohl, just now I saw that you did make a new commit. Thanks again!
